File: Bars.Gkh/libs/B4/model/SupplyResourceOrg.js
Ext.define('B4.model.SupplyResourceOrg', {
    extend: 'B4.base.Model',
    idProperty: 'Id',
    requires: [ 'B4.enums.OrgStateRole',
                'B4.enums.GroundsTermination'],
    proxy: {
        type: 'b4proxy',
        controllerName: 'SupplyResourceOrg'
    },
    fields: [
        { name: 'Id', useNull: true },
        { name: 'Contragent', defaultValue: null },
        { name: 'OrgStateRole', defaultValue: 10 },
        { name: 'ContragentId' },
        { name: 'ContragentName' },
        { name: 'Description' },
        { name: 'ActivityGroundsTermination', defaultValue: 10 },
        { name: 'Municipality' },
        { name: 'Settlement' },
        { name: 'DateTermination' },
        { name: 'Inn' }
    ]
});

File: Bars.Gkh/Entities/SupplyResourceOrg/SupplyResourceOrg.cs
namespace Bars.Gkh.Entities
{
    using System;

    using Bars.Gkh.Enums;

    /// <summary>
    /// Поставщик коммунальных услуг
    /// </summary>
    public class SupplyResourceOrg : BaseGkhEntity
    {
        /// <summary>
        /// Контрагент
        /// </summary>
        public virtual Contragent Contragent { get; set; }

        /// <summary>
        /// Описание
        /// </summary>
        public virtual string Description { get; set; }

        /// <summary>
        /// Статус
        /// </summary>
        public virtual OrgStateRole OrgStateRole { get; set; }

        /// <summary>
        /// Наименование контрагента (Не хранимое)
        /// </summary>
        public virtual string ContragentName { get; set; }

        /// <summary>
        /// Основание прекращения деятельности
        /// </summary>
        public virtual GroundsTermination ActivityGroundsTermination { get; set; }

        /// <summary>
        /// Дата прекращения деятельности
        /// </summary>
        public virtual DateTime? DateTermination { get; set; }

        /// <summary>
        /// Примечание прекращения деятельности
        /// </summary>
        public virtual string DescriptionTermination { get; set; }
    }
}
